### PR TITLE
chore: use loaded module reference

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
What:
 - Medusa App can use the reference of already imported mmodules.
Why:
 - Bundled apps need to export all the modules they will use and pass them as reference. Otherwise the tree shaking mechanism will remove them from the final bundle.

### CODE REVIEW

File: packages/pricing/src/index.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { ModulesSdkUtils } from "@medusajs/utils"
 
 export default moduleDefinition
 
+
```
Comment (adrien2p) [inline packages/pricing/src/index.ts]: we can probably revert that file 😄

File: packages/product/src/index.ts [hunk 2]
```diff
@@ -18,8 +19,9 @@ export const revertMigration = ModulesSdkUtils.buildRevertMigrationScript(
   migrationScriptOptions
 )
 
+
 export * from "./initialize"
-export * from "./types"
 export * from "./loaders"
 export * from "./models"
 export * from "./services"
+export * from "./types"
```
Comment (adrien2p) [inline packages/product/src/index.ts:14]: we can probably revert that file 😄

File: .changeset/fuzzy-tips-clap.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"@medusajs/modules-sdk": patch
+"@medusajs/types": patch
```
Comment (adrien2p) [inline .changeset/fuzzy-tips-clap.md:3]: todo: missing medusa package

**Review by adrien2p — APPROVED** [review]
LGTM 🔥

File: .changeset/fuzzy-tips-clap.md (hunk 3 above)
Comment (carlos-r-l-rodrigues) [reply .changeset/fuzzy-tips-clap.md:3]: technically that wasn't changed, so I think it is fine to keep that out.

Comment (adrien2p) [reply .changeset/fuzzy-tips-clap.md:3]: the resolve type did change to accept ModuleExports now, shouldn't this be released?

File: packages/types/src/modules-sdk/index.ts [hunk 4]
```diff
@@ -227,14 +227,20 @@ export declare type ModuleJoinerRelationship = JoinerRelationship & {
...
   runMigrations?(
-    options: LoaderOptions,
+    options: LoaderOptions<any>,
     moduleDeclaration?: InternalModuleDeclaration
   ): Promise<void>
   revertMigration?(
-    options: LoaderOptions,
+    options: LoaderOptions<any>,
     moduleDeclaration?: InternalModuleDeclaration
   ): Promise<void>
```
Comment (adrien2p) [inline packages/types/src/modules-sdk/index.ts:245]: we will need to update all the modules, they are all exporting it from the entry point, and none are proving these functions from the moduleDefinition. wdyt? and this (loadModuleMigrations) will change as well right from the load internals?

Comment (carlos-r-l-rodrigues) [reply packages/types/src/modules-sdk/index.ts:245]: good catch. updated them all

**Comment by carlos-r-l-rodrigues** [comment]
@olivermrbl quick cleanup to normalize what the modules are exporting + Enable MedusaApp to receive the reference of the module instead of it's path.

File: .changeset/fuzzy-tips-clap.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+"@medusajs/authetication": patch
+"@medusajs/cart": patch
+"@medusajs/customer": patch
+"@medusajs/payment": patch
+"@medusajs/pricing": patch
+"@medusajs/product": patch
+"@medusajs/promotion": patch
```
Comment (riqwan) [inline .changeset/fuzzy-tips-clap.md]:
```suggestion
```

promotion isn't release yet, you can skip this one

File: .changeset/fuzzy-tips-clap.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+---
+"@medusajs/modules-sdk": patch
+"@medusajs/types": patch
+"@medusajs/authetication": patch
+"@medusajs/cart": patch
+"@medusajs/customer": patch
+"@medusajs/payment": patch
```
Comment (riqwan) [inline .changeset/fuzzy-tips-clap.md]:
```suggestion
```

Same with these as well, but gotta check with the module owners.

**Review by riqwan — APPROVED** [review]
lgtm!

File: .changeset/fuzzy-tips-clap.md (hunk 6 above)
Comment (carlos-r-l-rodrigues) [reply .changeset/fuzzy-tips-clap.md]: good catch... will remove them

**Comment by carlos-r-l-rodrigues** [comment]
@olivermrbl can I merge this one?

**Comment by olivermrbl** [comment]
Yes, for sure!

**Pull request merged**